recipes/mysql-concpp/all/conanfile.py

```diff
@@ -0,0 +1,207 @@
+            replace_in_file(self, os.path.join(self.source_folder, "install_layout.cmake"),
+                                "set(LIB_NAME_STATIC \"${LIB_NAME}-mt\")",
+                                "set(LIB_NAME_STATIC \"${LIB_NAME_STATIC}-mt\")",
+                                strict=False)
+
+        # ZSTD patch
+        replace_in_file(self, os.path.join(self.source_folder, "cdk", "extra", "zstd", "CMakeLists.txt"),
+                            "enable_pic()",
+                            "enable_pic()\n"\
+                            "add_compile_definitions(ZSTD_DISABLE_ASM)",
+                            strict=False)
+
+        # Apple patches
+        if is_apple_os(self) and cross_building(self):
+            print(f"Building for {str(self.settings.arch)}")
+            patch = f"set(CMAKE_OSX_ARCHITECTURES \"{str(self.settings.arch)}\" CACHE INTERNAL \"\" FORCE)\n"
+
+            replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                                "PROJECT(MySQL_CONCPP)",
+                                f"PROJECT(MySQL_CONCPP)\n{patch}",
+                                strict=False)
+            # General patches
+            for lb in ["lz4", 'zlib', 'protobuf', 'zstd']:
+                replace_in_file(self, os.path.join(self.source_folder, "cdk", "extra", lb, "CMakeLists.txt"),
+                                    "enable_pic()",
+                                    f"enable_pic()\n{patch}",
+                                    strict=False)
+
+    def build(self):
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+        copy(self, "LICENSE", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+
+    def package_info(self):
+        # Set the VS version
+        if self.settings.os == "Windows":
+            msvc_toolset_version = self.settings.get_safe("compiler.toolset")
+            msvc_version = self.settings.get_safe("compiler.version")
+
+            if msvc_toolset_version:
+                vs_version = msvc_toolset_version[1:3]
+            elif msvc_version:
+                vs_version = msvc_version[:2]
+                if vs_version == "18":
+                    vs_version = "12"
+                elif vs_version == "19":
+                    vs_version = "14"
+            else:
+                self.output.warn("MSVC_TOOLSET_VERSION or MSVC_VERSION not defined")
+                vs_version = "unknown"
+
+            self.output.info(f"VS Version: {vs_version}")
+            self.vs_version = vs_version
+            self.vs = f"vs{vs_version}"
+        else:
+            self.vs = ""
+
+        template_dirs = ["lib64", "lib64/debug", "lib", "lib/debug"]
+        self.cpp_info.libdirs = template_dirs if not self.vs else [f"{lib}/{self.vs}" for lib in template_dirs]
```

Comment:
```suggestion
    @property
    def _vs_version(self):
        msvc_toolset_version = self.settings.get_safe("compiler.toolset")
        msvc_version = self.settings.get_safe("compiler.version")

        if msvc_toolset_version:
            vs_version = msvc_toolset_version[1:3]
        elif msvc_version:
            vs_version = msvc_version[:2]
            if vs_version == "18":
                vs_version = "12"
            elif vs_version == "19":
                vs_version = "14"
        else:
            self.output.warn("MSVC_TOOLSET_VERSION or MSVC_VERSION not defined")
            vs_version = "unknown"

        self.output.info(f"VS Version: {vs_version}")
        return f"vs{vs_version}"

    def package_info(self):
        template_dirs = ["lib64", os.path.join("lib64", "debug"), "lib", os.path.join("lib", "debug")]
        if is_msvc(self):
            vs_version = self._vs_version
            self.cpp_info.libdirs = [os.path.join(lib, vs_version) for lib in template_dirs]
        else:
            self.cpp_info.libdirs = template_dirs
```
I suggest moving `self.vs` logic into a separate function and using `os.path.join()`.